invenio_records_permissions/permissions/base.py

```diff
@@ -6,99 +6,108 @@
+
+from itertools import chain
+
 from flask import current_app
 from invenio_access import Permission
+
 from ..generators import Deny
 
+# Where can a property be used?
+#
+# |    Action   | need | excludes | query_filters |
+# |-------------|------|----------|---------------|
+# |    create   |   x  |     x    |               |
+# |-------------|------|----------|---------------|
+# |     list    |   x  |     x    |               |
+# |-------------|------|----------|---------------|
+# |     read    |   x  |     x    |        x      |
+# |-------------|------|----------|---------------|
+# | read files  |   x  |     x    |               |
+# |-------------|------|----------|---------------|
+# |    update   |   x  |     x    |               |
+# |-------------|------|----------|---------------|
+# |    delete   |   x  |     x    |               |
+# |-------------|------|----------|---------------|
+#
 
-class _PermissionConfig(object):
 
-    # Deny all by default
-    can_create = [Deny]
-    can_list = [Deny]
-    can_read = [Deny]
-    can_update = [Deny]
-    can_delete = [Deny]
-
-    @classmethod
-    def get_permission_list(cls, action):
-        if action == 'create':
-            return cls.can_create
-        elif action == 'list':
-            return cls.can_list
-        elif action == 'read':
-            return cls.can_read
-        elif action == 'update':
-            return cls.can_update
-        elif action == 'delete':
-            return cls.can_delete
-
-        current_app.logger.error("Unkown action {action}.".format(
-            action=action))
-        return []
+class BasePermissionPolicy(Permission):
+    """
+    BasePermissionPolicy to inherit from.
 
+    The class defines the overall policy and the instance encapsulates the
+    permissions for an *action* *over* a set of objects.
 
-# Where can a property be used?
-#
-# |    Action   | need | excludes | query_filter |
-# |-------------|------|----------|--------------|
-# |    create   |   x  |     x    |              |
-# |-------------|------|----------|--------------|
-# |     list    |   x  |     x    |              |
-# |-------------|------|----------|--------------|
-# |     read    |   x  |     x    |       x      |
-# |-------------|------|----------|--------------|
-# | read files  |   x  |     x    |              |
-# |-------------|------|----------|--------------|
-# |    update   |   x  |     x    |              |
-# |-------------|------|----------|--------------|
-# |    delete   |   x  |     x    |              |
-# |-------------|------|----------|--------------|
-#
+    TODO: Recognize a PermissionPolicy class in other modules instead of
+          individual factory functions to lessen the configuration burden.
+    """
 
+    # Deny all by default
+    can_list = [Deny()]
+    can_create = [Deny()]
+    can_read = [Deny()]
+    can_update = [Deny()]
+    can_delete = [Deny()]
+
+    def __init__(self, action, **over):
```

Comment:
Shall we pass `*args` and ``**kwargs`` and then look for the `over` key in one of them? Just to give more flexibility in the future.

Reply:
I wanted `**over` here for more clarity in our specific usage (`**kwargs` is too generic): a PermissionPolicy controls access of an **action** **over** resources. `**over` is very flexible: any keyword argument can be passed. But if the constructor's signature is too constraining for a developer, she can just inherit `BasePermissionPolicy` and redefine the `__init__` however she wants. 

As for `*args` I am not a fan of them for the reason we saw in `invenio_access`.
I will leave them as is for now.